#154953780 Show exercise info

nyandiekaFelix opened the pull request:
#### What does this PR do?
 Add a read-only API endpoint that shows all infos for an exercise

#### Description of Task to be completed?
There should be a new read-only API endpoint that collects all the information of an exercise

Added endpoint:
```
GET /api/v2/exercise
GET /api/v2/exercise/<id>
```
#### How should this be manually tested?
- After setting up the project based on the instructions in the README,
run server with `python manage.py runserver`.
- Using postman , enter url `http://localhost:8000/api/v2/exerciseinfo/<id>`

#### What are the relevant pivotal tracker stories?
[154953780](https://www.pivotaltracker.com/story/show/154953780)

#### Screenshots 

<img width="1440" alt="screen shot 2018-02-19 at 11 47 05" src="https://user-images.githubusercontent.com/15908547/36369006-beee5566-156a-11e8-83ea-570216156b3e.png">

muchai-mercy commented on wger/urls.py:
```diff
@@ -103,6 +103,7 @@
 
 # Exercises app
 router.register(r'exercise', exercises_api_views.ExerciseViewSet, base_name='exercise')
+router.register(r'exerciseinfo', exercises_api_views.ExerciseInfoViewSet, base_name='exerciseinfo')
```
Could you register the route as we mentioned, to take `exercise/<id>`

muchai-mercy commented:
You may need to re-run `pycodestyle wger/` because of the tests failing? @nyandiekaFelix

nyandiekaFelix commented:
@Mercy-Muchai  The Travis build is passing now. I fixed the coding style errors.

muchai-mercy commented on wger/exercises/api/views.py:
```diff
@@ -58,6 +59,45 @@ def perform_create(self, serializer):
+    exercise = get_object_or_404(Exercise, pk=id)
+
+    exercise_data['exercise'] = exercise
+    # exercise_data['languages'] = languages
```
Do we need this?

muchai-mercy commented on wger/exercises/api/views.py:
```diff
@@ -58,6 +59,45 @@ def perform_create(self, serializer):
+    images = ['/media/' + str(image.image)
+              for image in exercise.exerciseimage_set.accepted()]
+    exercise_data['images'] = images
+    print(exercise_data)
```
I don't think we need this print either

muchai-mercy commented on .gitignore:
```diff
@@ -58,10 +58,9 @@ target/
+
 .DS_Store
-venv-django
 .vscode/
```
Not sure you merged this file correct. Compare

muchai-mercy commented on wger/urls.py:
```diff
@@ -222,6 +222,9 @@
         nutrition_api_views.search,
         name='ingredient-search'),
     url(r'^api/v2/', include(router.urls)),
+    url(r'^api/v2/exercise/(?P<id>\d+)/$',
```
Any changes on this @nyandiekaFelix ?

nyandiekaFelix replied:
I changed the parameter `name='exerciseinfo'` to `name='exercise'`. Anything else I need to change? @Mercy-Muchai

muchai-mercy commented:
@nyandiekaFelix squash commits so it's ready for merge

nyandiekaFelix commented:
@Mercy-Muchai  the commits have been squashed.

muchai-mercy commented:
@nyandiekaFelix Fix failing test builds

nyandiekaFelix committed:
[Feature #154953780] Add exercise info url endpoint:

                    - Add GET only endpoint method
                    - Register exercise url to accept ID
                    - Update information displayed on HTML

nyandiekaFelix commented:
@Mercy-Muchai  Build is passing now.

JoyyToo reviewed (approved):
LGTM

muchai-mercy reviewed (approved):
LGTM
